packages/grid/src/TextCellEditor.tsx

```diff
@@ -50,15 +53,26 @@ export function TextCellEditor<T>(props: TextCellEditorProps<T>) {
   };
 
   useEffect(() => {
-    if (
-      inputRef.current &&
-      !firstRenderSelectionRef.current &&
-      editorText === column!.info.props.getValue!(row!.data)
-    ) {
+    if (inputRef.current && !initialSelectionRef.current) {
       inputRef.current.select();
+      initialSelectionRef.current = true;
     }
-    firstRenderSelectionRef.current = true;
-  }, [column, editorText, row]);
+  }, [inputRef.current]);
+
+  useEffect(() => {
+    const input = inputRef.current;
+    const focusOut = (event: FocusEvent) => {
+      if (!input?.contains(event.target as Node)) {
+        endEditMode(editorText);
+      }
+    };
+
+    document?.addEventListener("mousedown", focusOut, true);
```

Comment:
Do we need some comment around capture v.s. race condition?

Reply:
Added :)